external-crates/move/crates/move-analyzer/src/symbols.rs

```diff
@@ -2767,22 +2813,17 @@ pub fn type_def_loc(
     match t {
         Type_::Ref(_, r) => type_def_loc(mod_outer_defs, r),
         Type_::Apply(_, sp!(_, TypeName_::ModuleType(sp!(_, mod_ident), struct_name)), _) => {
-            find_datatype(mod_outer_defs, mod_ident, &struct_name.value())
+            let mod_ident_str = expansion_mod_ident_to_map_key(mod_ident);
+            mod_outer_defs
+                .get(&mod_ident_str)
+                .map(|mod_defs| find_datatype(mod_defs, &struct_name.value()))
+                .flatten()
         }
         _ => None,
     }
 }
 
-fn find_datatype(
-    mod_outer_defs: &BTreeMap<String, ModuleDefs>,
-    mod_ident: &ModuleIdent_,
-    datatype_name: &Symbol,
-) -> Option<Loc> {
-    let mod_ident_str = expansion_mod_ident_to_map_key(mod_ident);
-    let mod_defs = match mod_outer_defs.get(&mod_ident_str) {
-        Some(v) => v,
-        None => return None,
-    };
+pub fn find_datatype(mod_defs: &ModuleDefs, datatype_name: &Symbol) -> Option<Loc> {
```

Comment:
This is the "unrelated" refactoring mentioned in another comment. It wasn't the cleanest to pass `mod_outer_defs` and retrieve `mod_defs` out of it the second time when already having `mod_defs` available in the caller.